Projects/UOContent/Mobiles/AI/BaseAI.cs

```diff
@@ -1834,242 +1711,274 @@ public virtual bool DoOrderTransfer()
             {
-                var args = $"{to.Name}\t{from.Name}\t ";
-
-                from.SendLocalizedMessage(
-                    1043250,
-                    args
-                ); // The pet refuses to be transferred because it will not obey you sufficiently.~3_BLANK~
-                to.SendLocalizedMessage(
-                    1043251,
-                    args
-                ); // The pet will not accept you as a master because it does not trust ~2_NAME~.~3_BLANK~
+                SendTransferRefusalMessages(from, to, 1043250, 1043251);
+                // 1043250: The pet refuses to be transferred because it will not obey you sufficiently.~3_BLANK~
+                // 1043251: The pet will not accept you as a master because it does not trust ~2_NAME~.~3_BLANK~
             }
-            else if (TransferItem.IsInCombat(m_Mobile))
+            else if (IsInCombatState())
             {
-                from.SendMessage("You may not transfer a pet that has recently been in combat.");
-                to.SendMessage("The pet may not be transferred to you because it has recently been in combat.");
+                from.SendLocalizedMessage(1042561);
+                // 1042561: You may not transfer a pet that has recently been in combat.
+                to.SendLocalizedMessage(1042562);
+                // 1042562: The pet may not be transferred to you because it has recently been in combat.
             }
             else
             {
-                NetState fromState = from.NetState, toState = to.NetState;
+                HandleTransferOrder(from, to);
+            }
 
-                if (fromState != null && toState != null)
-                {
-                    if (from.HasTrade)
-                    {
-                        from.SendLocalizedMessage(1010507); // You cannot transfer a pet with a trade pending
-                    }
-                    else if (to.HasTrade)
-                    {
-                        to.SendLocalizedMessage(1010507); // You cannot transfer a pet with a trade pending
-                    }
-                    else
-                    {
-                        Container c = fromState.AddTrade(toState);
-                        c.DropItem(new TransferItem(m_Mobile));
-                    }
-                }
+            if (m_Mobile.ControlOrder != OrderType.Stay)
+            {
+                m_Mobile.ControlTarget = null;
+                m_Mobile.ControlOrder = OrderType.Stay;
             }
+
+            return true;
         }
 
-        m_Mobile.ControlTarget = null;
-        m_Mobile.ControlOrder = OrderType.Stay;
+        return false;
+    }
 
-        return true;
+    private bool IsInCombatState()
+    {
+        return m_Mobile.Combatant != null || m_Mobile.Aggressors.Count > 0 ||
+        m_Mobile.Aggressed.Count > 0 || Core.TickCount < m_Mobile.NextCombatTime;
+    }
+    
+    private bool IsValidTransferRequest(Mobile from, Mobile to)
```

Comment:
I see duplicate validity check logic, we should consolidate these functions to one?